token/cli/src/main.rs

```diff
@@ -499,77 +515,62 @@ async fn command_set_interest_rate(
-            .program_client
-            .get_minimum_balance_for_rent_exemption(Account::LEN)
-            .await?
-    } else {
-        0
-    };
+    let token = token_client_from_config(config, &token_pubkey);
+    let mut extensions = vec![];
 
-    let mint_info = config.get_mint_info(&token, None).await?;
-    let (account, system_account_ok, instructions) = if let Some(account) = maybe_account {
-        println_display(config, format!("Creating account {}", account));
-        (
-            account,
-            false,
-            vec![
-                system_instruction::create_account(
-                    &config.fee_payer.pubkey(),
-                    &account,
-                    minimum_balance_for_rent_exemption,
-                    Account::LEN as u64,
-                    &mint_info.program_id,
-                ),
-                initialize_account(&mint_info.program_id, &account, &token, &owner)?,
-            ],
-        )
+    let (account, is_associated) = if let Some(account) = maybe_account {
+        (account, false)
     } else {
-        let account =
-            get_associated_token_address_with_program_id(&owner, &token, &mint_info.program_id);
-        println_display(config, format!("Creating account {}", account));
-        (
-            account,
-            true,
-            vec![create_associated_token_account(
-                &config.fee_payer.pubkey(),
-                &owner,
-                &token,
-                &mint_info.program_id,
-            )],
-        )
+        (token.get_associated_token_address(&owner), true)
     };
 
+    println_display(config, format!("Creating account {}", account));
+
     if !config.sign_only {
-        if let Some(account_data) = config
-            .rpc_client
-            .get_account_with_commitment(&account, config.rpc_client.commitment())
-            .await?
-            .value
-        {
-            if !(account_data.owner == system_program::id() && system_account_ok) {
+        if let Some(account_data) = config.program_client.get_account(account).await? {
+            if account_data.owner != system_program::id() || !is_associated {
                 return Err(format!("Error: Account already exists: {}", account).into());
             }
         }
     }
 
-    let tx_return = handle_tx(
-        &CliSignerInfo {
-            signers: bulk_signers,
-        },
-        config,
-        false,
-        minimum_balance_for_rent_exemption,
-        instructions,
-    )
-    .await?;
+    if immutable_owner {
+        if config.program_id == spl_token::id() {
+            return Err(format!(
+                "Specified --immutable, but token program {} does not support the extension",
+                config.program_id
+            )
+            .into());
+        } else if is_associated {
+            println_display(
+                config,
+                "Note: --immutable specified, but Token-2022 ATAs are always immutable".to_string(),
```

Comment:
take it or leave it:
```suggestion
                "Note: --immutable specified, but Token-2022 ATAs are always immutable, ignoring".to_string(),
```